### PR TITLE
docs: readme update

### PR DESCRIPTION
**Pull request by michalbali256**

### CODE REVIEW

File: clients/vscode-hlasmplugin/README.md [hunk 1]
```diff
@@ -123,17 +123,22 @@ The following example specifies that GROUP1 is used when working with `source_co
...
     },
-  ]
+  ],
+  "alwaysRecognize" : ["*.hlasm", "libs/*.asm"]
 }
 ```
 If you have the two configuration files configured as above and invoke the MAC1 macro from `source_code.hlasm`, the folder `ASMMAC/` in the current workspace is searched for a file with the exact name "MAC1". If that search is unsuccessful the folder `C:/SYS.ASMMAC` is searched. If that search is unsuccessful an error displays that the macro does not exist.
 
-The program field in `pgm_conf.json` supports regular expressions, for example:
+There is also option `alwaysRecognize` which takes array of wildcards. It allows you to configure two things:
```
Comment (asmelko) [inline clients/vscode-hlasmplugin/README.md:132]:
```suggestion
There is also the option `alwaysRecognize` which takes an array of wildcards. It allows you to configure two things:
```

File: clients/vscode-hlasmplugin/README.md [hunk 2]
```diff
@@ -123,17 +123,22 @@ The following example specifies that GROUP1 is used when working with `source_co
...
+- All files matching these wildcards will be always recognized as HLASM files. 
+- Macros and copy files will be recognized even if their file names does not match the identifier written into source code. For example, macro `IF` would be found even if it was defined in file `IF.hlasm`.
```
Comment (asmelko) [inline clients/vscode-hlasmplugin/README.md:134]:
```suggestion
- Macro and copy files will be recognized even if their file names do not match the identifier used in a source code. For example, macro `IF` would be found even if it was defined in file `IF.hlasm`.
```

File: clients/vscode-hlasmplugin/README.md [hunk 3]
```diff
@@ -123,17 +123,22 @@ The following example specifies that GROUP1 is used when working with `source_co
...
+
+The program field in `pgm_conf.json` supports wildcards, for example:
 ```
 {
   "pgms": [
     {
-      "program": ".*",
+      "program": "*",
```
Comment (asmelko) [inline clients/vscode-hlasmplugin/README.md:141]: a bit vague example, don't you think?

File: clients/vscode-hlasmplugin/README.md [hunk 4]
```diff
@@ -123,17 +123,22 @@ The following example specifies that GROUP1 is used when working with `source_co
...
-  ]
+  ],
+  "alwaysRecognize" : ["*.hlasm", "libs/*.asm"]
 }
 ```
 If you have the two configuration files configured as above and invoke the MAC1 macro from `source_code.hlasm`, the folder `ASMMAC/` in the current workspace is searched for a file with the exact name "MAC1". If that search is unsuccessful the folder `C:/SYS.ASMMAC` is searched. If that search is unsuccessful an error displays that the macro does not exist.
 
-The program field in `pgm_conf.json` supports regular expressions, for example:
+There is also option `alwaysRecognize` which takes array of wildcards. It allows you to configure two things:
+- All files matching these wildcards will be always recognized as HLASM files.
```
Comment (marcel1hruska) [inline clients/vscode-hlasmplugin/README.md:133]:
```suggestion
- All files matching these wildcards will always be recognized as HLASM files. 
```

File: clients/vscode-hlasmplugin/README.md (hunk 3 above)
Comment (michalbali256) [reply clients/vscode-hlasmplugin/README.md:141]: that example should only demonstrate how to get the easiest setup. What do you suggest?

File: clients/vscode-hlasmplugin/README.md (hunk 2 above)
Comment (marcel1hruska) [reply clients/vscode-hlasmplugin/README.md]: This does not really explains the functionality. I would recommend to reformulate it, e.g. "If an extension wildcard is defined, all macro and copy files with such extension may be used in the source code. For example, with the extension wildcard '*.hlasm', a user may add macro __MAC__ to his source code even if it is in a file called __Mac.hlasm__."

File: clients/vscode-hlasmplugin/README.md (hunk 3 above)
Comment (asmelko) [reply clients/vscode-hlasmplugin/README.md:141]: maybe something that demonstrates wildcards more, `*.asm`. I agree with your response though, so it's OK for me.

Comment (marcel1hruska) [reply clients/vscode-hlasmplugin/README.md:141]: common case would probably be a folder with open source files, e.g. pgms/*

Comment (michalbali256) [reply clients/vscode-hlasmplugin/README.md:141]: what about "ASMPGM/*" ?

**Pull request merged**